rfcs/rfc00XX.md

```diff
@@ -0,0 +1,257 @@
+# Built-in lexical exporting
+
+## Preamble
+
+    Author:  Curtis Poe <curtis.poe@gmail.com>
+    Sponsor:
+    ID:      XX
+    Status:  Draft
+
+## Abstract
+
+This RFC proposes adding built-in lexical exporting via an `:export`
+attribute.
+
+## Motivation
+
+To eliminate the need for the myriad different CPAN modules for exporting
+functions. This RFC is only for functions, not variables.
+
+## Rationale
+
+There are many exporter modules available. The core `Exporter` is the best
+known, and has the curious feature of allowing procedural modules to use
+inheritance to implement exporting via package variables. It has a clumsy
+interface and many alternatives arose:
+
+* [Sub::Exporter](https://metacpan.org/pod/Sub::Exporter)
+* [Exporter::Lite](https://metacpan.org/pod/Exporter::Lite)
+* [Exporter::Easy](https://metacpan.org/pod/Exporter::Easy)
+* [Exporter::Tiny](https://metacpan.org/pod/Exporter::Tiny)
+
+And more, including many modules which manually implement exporting directly
+in the `import` method.
+
+There are also many attribute-based exporters:
+
+* [Exporter::Simple](https://metacpan.org/pod/Exporter::Simple)
+* [Perl6::Export](https://metacpan.org/pod/Perl6::Export) (not really attributes)
+* [Perl6::Export::Attrs](https://metacpan.org/pod/Perl6::Export::Attrs)
+
+And so on.
+
+## Specification
+
+Because lower-case attribute names are reserved for Perl's future expansion,
+we do not need a feature guard. Instead, we can simply `use v5.XX` to pull in
+the new feature.
+
+    package My::Module {
+        use v5.38;
+
+        sub bar  :export            {...} # @EXPORT_OK
+        sub baz  :export(:strings)  {...} # %EXPORT_TAGS
+        sub bay  :export(:strings)  {...} # %EXPORT_TAGS
+        sub quux :export(:ALWAYS)   {...} # @EXPORT
+        sub whee                    {...} # not exported
+    }
+
+The above can be written as this:
+
+    package My::Module {
+        use base 'Exporter';
+
+        our @EXPORT    = qw(quux);
+        our @EXPORT_OK = qw(
+            bar
+            baz
+            bay
+        )
+        our %EXPORT_TAGS = qw(
+            'all'     => \@EXPORT_OK,
+            'strings' => qw(bay baz),
+        );
+
+        sub bar  {...}
+        sub baz  {...}
+        sub bay  {...}
+        sub quux {...}
+        sub whee {...}
+    }
+
+
+Imported functions do not remain in the namespace, removing the need for
```

Comment:
Better wording here might be:

"The requested functions are imported lexically, rather than appearing in the package symbol table. This means that code within that lexical scope can see and call them as normal, but they are not visible from outside - such as from other callers, or as object or class methods. This removes the need for `namespace::clean` and related modules."